command/agent/job_endpoint_test.go

```diff
@@ -1964,9 +1964,11 @@ func TestHTTP_JobDispatch(t *testing.T) {
 			},
 		}
 		var resp structs.JobRegisterResponse
-		if err := s.Agent.RPC("Job.Register", &args, &resp); err != nil {
-			t.Fatalf("err: %v", err)
+		err := s.Agent.RPC("Job.Register", &args, &resp)
+		if err != nil {
+			t.Logf("err: %v", err)
 		}
```

Comment:
If we have an unexpected error, `must.NoError` will report it, so we don't need this bit anymore.

```suggestion
```